sdk/keyvault/internal/challenge_policy.go

```diff
@@ -170,17 +171,27 @@ func (k *KeyVaultChallengePolicy) findScopeAndTenant(resp *http.Response) error
 	}
 
 	k.tenantID = parseTenant(vals["authorization"])
-	if scope, ok := vals["scope"]; ok {
-		k.scope = &scope
-	} else if resource, ok := vals["resource"]; ok {
-		if !strings.HasSuffix(resource, "/.default") {
-			resource += "/.default"
-		}
-		k.scope = &resource
-	} else {
+	scope := ""
+	if v, ok := vals["scope"]; ok {
+		scope = v
+	} else if v, ok := vals["resource"]; ok {
+		scope = v
+	}
+	if scope == "" {
 		return &challengePolicyError{err: errors.New("could not find a valid resource in the WWW-Authenticate header")}
 	}
-
+	// the challenge resource's host must match the requested vault's host
+	parsed, err := url.Parse(scope)
+	if err != nil {
+		return &challengePolicyError{err: fmt.Errorf(`invalid challenge resource "%s": %v`, scope, err)}
+	}
+	if !strings.HasSuffix(req.URL.Host, "."+parsed.Host) {
+		return &challengePolicyError{err: fmt.Errorf(`the challenge resource "%s" doesn't match the requested domain`, scope)}
+	}
+	if !strings.HasSuffix(scope, "/.default") {
+		scope += "/.default"
+	}
```

Comment:
Why not just always add "/.default" when checking for "resource" in condition block on line 177?

Reply:
I did this to simplify reporting the resource value in the error message. If the value were `https://foo`, I don't want the error to include `https://foo/.default`.